apps/web/src/components/auth/githubUtils.ts

```diff
@@ -0,0 +1,31 @@
+import { SignUpOriginEnum } from '@novu/shared';
+
+import { API_ROOT } from '../../config';
+
+export const buildGithubLink = ({ invitationToken }: { invitationToken?: string }) => {
+  const queryParams = new URLSearchParams();
+  queryParams.append('source', SignUpOriginEnum.WEB);
+  if (invitationToken) {
+    queryParams.append('invitationToken', invitationToken);
+  }
+
+  return `${API_ROOT}/v1/auth/github?${queryParams.toString()}`;
+};
+
+export const buildVercelGithubLink = ({
+  code,
+  next,
+  configurationId,
+}: {
+  code: string | null;
+  next: string | null;
+  configurationId: string | null;
+}) => {
+  const queryParams = new URLSearchParams();
+  queryParams.append('partnerCode', code ?? '');
+  queryParams.append('next', next ?? '');
+  queryParams.append('configurationId', configurationId ?? '');
+  queryParams.append('source', SignUpOriginEnum.VERCEL);
+
+  return `${API_ROOT}/v1/auth/github?${queryParams.toString()}`;
+};
```

Comment:
@BiswaViraj you know this code well, could you verify if that doesn't have any issues?

Reply:
It looks ok for me comparing with the existing code, but let's have @BiswaViraj checking it. 🙏🏻 
Maybe I wouldn't add the `queryParams` with empty string value if they are not defined for cleanliness when building the URL and to avoid any potential edge case bug if the query param is passed.

Reply:
This might create an issue, as those fields are essential for vercel integration